background/index.ts

```diff
@@ -82,6 +86,79 @@ async function main() {
     },
     ["userOpPool"]
   )
+
+  const fetchData = async () => {
+    const fetchUserOpsSent = async () => {
+      const timeout = 1500
+      console.log(`[background] fetch userOp sent every ${timeout} ms`)
+
+      const s = storage.get()
+      const nm = new NetworkStorageManager(storage)
+
+      const chainId = nm.getActive().chainId
+      const bundler = nm.getActive().bundler
+
+      const userOps = Object.values(s.userOpPool)
+      const sentUserOps = userOps.filter((userOp) => userOp.status === "Sent")
+
+      sentUserOps.forEach(async (sentUserOp) => {
+        const id = sentUserOp.id
+        const userOp = new UserOperation(sentUserOp.userOp)
+        const userOpHash = userOp.hash(sentUserOp.entryPointAddress, chainId)
+        await bundler.wait(userOpHash)
+
+        const userOpReceipt = await bundler.send<{
+          success: boolean
+          reason: string
+          receipt: {
+            transactionHash: HexString
+            blockHash: HexString
+            blockNumber: BigNumberish
+          }
+        }>({
+          method: BundlerRpcMethod.eth_getUserOperationReceipt,
+          params: [userOpHash]
+        })
+
+        if (userOpReceipt && userOpReceipt.success) {
+          const succeededUserOp: UserOperationLog = {
+            ...sentUserOp,
+            status: UserOperationStatus.Succeeded,
+            receipt: {
+              userOpHash: userOpHash,
+              transactionHash: userOpReceipt.receipt.transactionHash,
+              blockHash: userOpReceipt.receipt.blockHash,
+              blockNumber: number.toHex(userOpReceipt.receipt.blockNumber)
+            }
+          }
+          const state = { userOpPool: { [id]: succeededUserOp } }
+          storage.set(state, { override: false })
+        }
+
+        if (userOpReceipt && !userOpReceipt.success) {
+          const succeededUserOp: UserOperationLog = {
+            ...sentUserOp,
+            status: UserOperationStatus.Failed,
+            receipt: {
+              userOpHash: userOpHash,
+              transactionHash: userOpReceipt.receipt.transactionHash,
+              blockHash: userOpReceipt.receipt.blockHash,
+              blockNumber: number.toHex(userOpReceipt.receipt.blockNumber),
+              errorMessage: userOpReceipt.reason
+            }
+          }
+          const state = { userOpPool: { [id]: succeededUserOp } }
+          storage.set(state, { override: false })
+        }
+      })
+
+      setTimeout(fetchUserOpsSent, timeout)
+    }
+    // First call
+    await fetchUserOpsSent()
+  }
+
+  await fetchData()
```

Comment:
I think we don't need this function, unwrap it and directly call `fetchUserOpsSent`